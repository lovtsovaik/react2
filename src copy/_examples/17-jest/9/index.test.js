// Core
import { v4 } from 'uuid';

class User {
    constructor(name) {
        this.name = name;
        this.created = new Date();
        this.id = v4(); // Уникальный идентификатор
    }
}

// Будет ломаться каждый раз
// test('«User» constructor should match its snapshot', () => {
//     expect(new User('Jack')).toMatchSnapshot();
// });

test('«User» constructor should match its snapshot', () => {
    expect(new User('Jack')).toMatchSnapshot({
        created: expect.any(Date),
        id:      expect.any(String),
    });
});
