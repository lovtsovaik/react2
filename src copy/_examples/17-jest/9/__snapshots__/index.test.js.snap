// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`«User» constructor should match its snapshot 1`] = `
Object {
  "created": Any<Date>,
  "id": Any<String>,
  "name": "Jack",
}
`;
